File: TinyScript/java/src/test/java/commons/AlphabetHelperTest.java
package commons;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

/**
 * @author MFine
 * @version 1.0
 * @date 2021/7/5 23:46
 **/
class AlphabetHelperTest {

    @Test
    public void test(){
        assertTrue(AlphabetHelper.isLetter('a'));
        assertFalse(AlphabetHelper.isLetter('*'));
        assertTrue(AlphabetHelper.isLetter('a'));
        assertTrue(AlphabetHelper.isLiteral('_'));
        assertTrue(AlphabetHelper.isLiteral('9'));
        assertFalse(AlphabetHelper.isLiteral('*'));
        assertFalse(AlphabetHelper.isLetter('*'));
        assertTrue(AlphabetHelper.isNumber('1'));
        assertTrue(AlphabetHelper.isNumber('9'));
        assertFalse(AlphabetHelper.isNumber('x'));
        assertTrue(AlphabetHelper.isOperator('&'));
        assertTrue(AlphabetHelper.isOperator('*'));
        assertTrue(AlphabetHelper.isOperator('+'));
        assertTrue(AlphabetHelper.isOperator('/'));
        assertTrue(AlphabetHelper.isOperator('='));
    }
}

File: TinyScript/js/src/test/commons/AlphabetHelper.test.ts
import AlphabetHelper from "../../lexer/AlphabetHelper";

describe('testAlphabetHelper',()=>{

    test('char check',()=>{
        expect(AlphabetHelper.isLetter('a')).toEqual(true);
        expect(AlphabetHelper.isLetter('A')).toEqual(true);
        expect(AlphabetHelper.isLetter('z')).toEqual(true);
        expect(AlphabetHelper.isLetter('Z')).toEqual(true);
        expect(AlphabetHelper.isLetter('1')).toEqual(false);

        expect(AlphabetHelper.isLiteral('0')).toEqual(true);
        expect(AlphabetHelper.isLiteral('9')).toEqual(true);
        expect(AlphabetHelper.isLiteral('_')).toEqual(true);
        expect(AlphabetHelper.isLiteral('a')).toEqual(true);
        expect(AlphabetHelper.isLiteral('Z')).toEqual(true);
        expect(AlphabetHelper.isLiteral(',')).toEqual(false);

        expect(AlphabetHelper.isNumber('0')).toEqual(true);
        expect(AlphabetHelper.isNumber('9')).toEqual(true);
        expect(AlphabetHelper.isNumber('p')).toEqual(false);

        expect(AlphabetHelper.isOperator('+')).toEqual(true);
        expect(AlphabetHelper.isOperator('-')).toEqual(true);
        expect(AlphabetHelper.isOperator('*')).toEqual(true);
        expect(AlphabetHelper.isOperator('/')).toEqual(true);
        expect(AlphabetHelper.isOperator('>')).toEqual(true);
        expect(AlphabetHelper.isOperator('<')).toEqual(true);
        expect(AlphabetHelper.isOperator('=')).toEqual(true);
        expect(AlphabetHelper.isOperator('!')).toEqual(true);
        expect(AlphabetHelper.isOperator('&')).toEqual(true);
        expect(AlphabetHelper.isOperator('|')).toEqual(true);
        expect(AlphabetHelper.isOperator('^')).toEqual(true);
        expect(AlphabetHelper.isOperator('%')).toEqual(true);
        expect(AlphabetHelper.isOperator('8')).toEqual(false);
        expect(AlphabetHelper.isOperator('z')).toEqual(false);

    })
})